types/lodash/index.d.ts

```diff
@@ -3610,108 +3611,79 @@ declare namespace _ {
          * @param values The property values.
          * @return Returns the new object.
          */
-        zipObject<TResult extends object>(
-            props: List<keyof TResult>,
-            values: List<TResult[keyof TResult]>
-        ): TResult;
+        zipObject<T>(
+            props: List<PropertyName>,
+            values: List<T>
+        ): Dictionary<T>;
 
         /**
          * @see _.zipObject
          */
         zipObject(
-            props: List<StringRepresentable>|List<List<any>>,
-            values?: List<any>
-        ): any;
-
-        /**
-         * @see _.zipObject
-         */
-        zipObject(): {};
+            props?: List<PropertyName>
+        ): Dictionary<undefined>;
 
         /**
          * This method is like _.zipObject except that it supports property paths.
          *
-         * @param props The property names.
+         * @param paths The property names.
          * @param values The property values.
          * @return Returns the new object.
          */
         zipObjectDeep(
-            props: List<StringRepresentable>|List<List<any>>,
+            paths?: List<PropertyPath>,
             values?: List<any>
-        ): any;
-
-        /**
-         * @see _.zipObjectDeep
-         */
-        zipObjectDeep(): {};
+        ): object;
```

Comment:
I don't like returning `object` because it's not assignable to pretty much anything. For example, this code will break:

```
interface Person {
    name: string;
    age: number;
}

const p: Person = _.zipObjectDeep(["name", "age"], ["Joe", 32]);
```

I would prefer to return `any` in the general case, and only return `object` if no arguments were provided.

Reply:
In general, type assertions should be explicit. So instead of

```ts
const p: Person = _.zipObjectDeep(["name", "age"], ["Joe", 32]);
```

it's better to write

```ts
const p = _.zipObjectDeep(["name", "age"], ["Joe", 32]) as Person;
```

Reply:
@aj-r Do you insist on using `any`? If so I'll change it, but I don't feel like it's the right choice.